.github/workflows/ci.yml

```diff
@@ -5,127 +5,68 @@ on:
     branches:
       - master
     paths-ignore:
-      - '**.md'
+      - "**.md"
   pull_request:
     branches:
       - master
     paths-ignore:
-      - '**.md'
+      - "**.md"
 
 env:
   TEST_ALL: false # Set this to true for testing the entire matrix
 
 jobs:
-  build:
+  ci:
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        rust: [stable, beta]
-        ruby_version: ["2.5.9", "2.6.10", "2.7.8"]
-        ruby_static: [false, true]
-        exclude:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        ruby: ["2.7", "3.0", "3.1", "3.2", "3.3", "head"]
+        rust: ["stable"]
+        include:
           - os: windows-latest
-            ruby_static: true
```

Comment:
In my experience with past Ruby version managers the Mac OS would be default install a statically built Ruby but Rutie ran best with Dynamic Linking.  The CI test suite accounted for that and permitted failures on the static build.  There were 3 failures that I remember for it.

What does the new CI test suite do for the differences between static and dynamic linking of Ruby? @goyox86